scripts/preinstall.js

```diff
@@ -0,0 +1,26 @@
+// Detect if run by NPM and fail
```

Comment:
@PierreBresson - CI really didn't seem to like the `only-allow` package. I have removed it in favour of this instead.
https://www.reddit.com/r/node/comments/j6xsdt/is_there_a_way_to_get_npm_to_throw_an_error_just/g81lkem/
![image](https://user-images.githubusercontent.com/52963980/95664454-b2a22180-0b93-11eb-9647-ff2d66fbc920.png)